modeltree.h

```diff
@@ -36,38 +36,50 @@
 namespace Asn1Acn {
 namespace Internal {
 
+class ProjectContentHandler;
+
 class ModelTree : public QObject
 {
     Q_OBJECT
 
     ModelTree();
     ~ModelTree() = default;
 
+    friend class ProjectContentHandler;
```

Comment:
Regrading the issue with ModelTree and ParsedDataStorage public API, After our discussion I have decided to leave the solution with ProjectContentHandler being friend for ModelTree and ParsedDataStorage. In this issue "Attorney-Client" pattern could be used, but I have decided not to implement it until component will become more mature, as it is sure that API in ModelTree and ParsedDataStorage will be changed.